controllers/ibmvpcmachinetemplate_controller.go

```diff
@@ -0,0 +1,115 @@
+
+	infrav1beta2 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta2"
+)
+
+// IBMVPCMachineTemplateReconciler reconciles a IBMVPCMachineTemplate object.
+type IBMVPCMachineTemplateReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *IBMVPCMachineTemplateReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&infrav1beta2.IBMVPCMachineTemplate{}).
+		Complete(r)
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ibmvpcmachinetemplates,verbs=get;list;watch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ibmvpcmachinetemplates/status,verbs=get;update;patch
+
+func (r *IBMVPCMachineTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling IBMVPCMachineTemplate")
+
+	var machineTemplate infrav1beta2.IBMVPCMachineTemplate
+	if err := r.Get(ctx, req.NamespacedName, &machineTemplate); err != nil {
+		log.Error(err, "Unable to fetch ibmvpcmachinetemplate")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	helper, err := patch.NewHelper(&machineTemplate, r.Client)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to init patch helper: %w", err)
+	}
+
+	capacity, err := getIBMVPCMachineCapacity(machineTemplate)
+	if err != nil {
+		log.Error(err, "Failed to get capacity from the ibmvpcmachine template")
+		return ctrl.Result{}, fmt.Errorf("failed to get capcity for machine template: %w", err)
+	}
+
+	log.V(3).Info("Calculated capacity for machine template", "capacity", capacity)
+	if !reflect.DeepEqual(machineTemplate.Status.Capacity, capacity) {
+		machineTemplate.Status.Capacity = capacity
+		if err := helper.Patch(ctx, &machineTemplate); err != nil {
+			if !apierrors.IsNotFound(err) {
+				log.Error(err, "Failed to patch machineTemplate")
+				return ctrl.Result{}, err
+			}
+		}
+	}
+	log.V(3).Info("Machine template status", "status", machineTemplate.Status)
+	return ctrl.Result{}, nil
+}
+
+func getIBMVPCMachineCapacity(machineTemplate infrav1beta2.IBMVPCMachineTemplate) (corev1.ResourceList, error) {
+	capacity := make(corev1.ResourceList)
+	// VPC profile will be in the below pattern
+	// PATTERN: "Alpha numeric value followed by '-' decimal which indicates cpu followed by 'x' followed by decimal which indicates memory and might be followed by GPU "
+	// Examples :
+	// 1.  VPC profile with GPU = 'gx2-32x256x2v100'
+	//     CPU = 32
+	//     Memory = 256G
+	//     Capacity = {cpu: "32", memory: "256G" }
+	// 2.  VPC profile without GPU = 'cx2d-16x32'
+	//     CPU = 16
+	//     Memory = 32G
+	//     Capacity = {cpu: "16", memory: "32G" }
+
+	regex := "\\S+[-]\\d+[x]\\d+\\S*$"
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile regular expression %s", regex)
+	}
+	if match := re.Match([]byte(machineTemplate.Spec.Template.Spec.Profile)); match {
```

Comment:
why don't we call the profile api to get more details about the memory and cpu information

https://cloud.ibm.com/apidocs/vpc/latest#get-instance-profile

Reply:
@KeerthanaAP have you explored on this?

Reply:
Yes @Prajyot-Parab I have modified the function with the profile api.